### ISSUE TITLE
[Issue]: Audio out of sync on webplayer while direct streaming

### ISSUE THREAD

**Issue by nicolai455** [jellyfin/jellyfin#9330]
### Please describe your bug

Hello,
recently i tried reencoding some files to hevc, while doing that i found a weird issue with media playback:

The audio is out of sync while direct streaming (transcoded audio from eac3 to aac, no video transcode) in the web player, but only after i skipped to any location in the video. When playing from the start the audio is in sync. 
Same problem occurs while transcoding other audio codecs like dts to aac.

The video playback works as expected on all other tested video players (Tested on VLC and the Android integrated player), and while direct playing, transcoding and just remuxing.
I saw the issue on chrome, brave and the andriod app, microsoft edge was playing eac3 native so it was no issue.

Steps to repruduce:
1) download h265 eac3 sample file from https://tdarr.io/samples/ and put it on jellyfin
2) use chrome browser to stream it
3) skip some time in the webplayer
4) hear the out of sync audio

Possible workarounds:
Use other media players
Force transcode by lowering bitrate or resolution

### Jellyfin Version

other

### if other:

10.8.9

### Environment

```markdown
- OS: Ubuntu Server 22.04.1 LTS
- Virtualization: none
- Clients: tested on chrome 110.0.5481.100, brave v1.48.164 and the Android app
- Browser: chrome, brave and the Android app (using web player)
- FFmpeg Version: 5.1.2-Jellyfin, also tried latest nightly N-109851-ge506ea3ce1
- Playback Method: Direct Stream
- Hardware Acceleration: tried nvenc and none
- Storage: NFS
```

### Jellyfin logs

```shell
[2023-02-16 19:44:50.434 +01:00] [INF] User policy for "user1234". EnablePlaybackRemuxing: True EnableVideoPlaybackTranscoding: True EnableAudioPlaybackTranscoding: True
[2023-02-16 19:44:50.434 +01:00] [INF] RemoteClientBitrateLimit: 45000000, RemoteIp: "172.71.160.73", IsInLocalNetwork: False
[2023-02-16 19:44:50.434 +01:00] [INF] StreamBuilder.BuildVideoItem( Profile="Anonymous Profile", Path="/mnt/share/data/media/series/TEST TDARR/Season 1000/sample__1080__libx265__eac3__30s__video (1).mkv", AudioStreamIndex=1, SubtitleStreamIndex=null ) => ( PlayMethod=Transcode, TranscodeReason=VideoCodecNotSupported, AudioCodecNotSupported ) "media:/videos/c436ed52-864e-4c74-3da0-48663a769944/stream.mp4?MediaSourceId=c436ed52864e4c743da048663a769944&VideoCodec=h264,hevc,vp9,av1,hevc&AudioCodec=aac,mp3,opus,flac,vorbis&AudioStreamIndex=1&VideoBitrate=44616000&AudioBitrate=384000&MaxFramerate=60&api_key=<token>&TranscodingMaxAudioChannels=2&RequireAvc=false&Tag=e7a97c08624d920faf5130864a666183&hevc-level=123&hevc-videobitdepth=8&hevc-profile=main&h264-profile=high,main,baseline,constrainedbaseline,high10&h264-rangetype=SDR&h264-level=52&h264-deinterlace=true&hevc-rangetype=SDR&hevc-deinterlace=true&vp9-rangetype=SDR,HDR10,HLG&av1-rangetype=SDR,HDR10,HLG&TranscodeReasons=VideoCodecNotSupported,%20AudioCodecNotSupported"
[2023-02-16 19:44:50.526 +01:00] [INF] Deleting partial stream file(s) "/var/lib/jellyfin/transcodes/cf405ef2dc715def79859b6b618286dc.mp4"
[2023-02-16 19:44:50.685 +01:00] [INF] "/usr/lib/jellyfin-ffmpeg/ffmpeg" "-analyzeduration 200M -ss 00:00:21.773 -fflags +genpts  -i file:\"/mnt/share/data/media/series/TEST TDARR/Season 1000/sample__1080__libx265__eac3__30s__video (1).mkv\" -map 0:0 -map 0:1 -map -0:s -codec:v:0 copy -map_metadata -1 -map_chapters -1 -threads 0 -codec:a:0 libfdk_aac -ac 2 -ab 384000 -f mp4 -movflags frag_keyframe+empty_moov -y \"/var/lib/jellyfin/transcodes/ca5df9fc4a63c1d06c1d29972f2b5e34.mp4\""
```

### FFmpeg logs

```shell
/videos/c436ed52-864e-4c74-3da0-48663a769944/stream.mp4

{"Protocol":0,"Id":"c436ed52864e4c743da048663a769944","Path":"/mnt/share/data/media/series/TEST TDARR/Season 1000/sample__1080__libx265__eac3__30s__video (1).mkv","EncoderPath":null,"EncoderProtocol":null,"Type":0,"Container":"mkv,webm","Size":6750543,"Name":"sample__1080__libx265__eac3__30s__video (1)","IsRemote":false,"ETag":"e7a97c08624d920faf5130864a666183","RunTimeTicks":300160000,"ReadAtNativeFramerate":false,"IgnoreDts":false,"IgnoreIndex":false,"GenPtsInput":false,"SupportsTranscoding":true,"SupportsDirectStream":true,"SupportsDirectPlay":true,"IsInfiniteStream":false,"RequiresOpening":false,"OpenToken":null,"RequiresClosing":false,"LiveStreamId":null,"BufferMs":null,"RequiresLooping":false,"SupportsProbing":true,"VideoType":0,"IsoType":null,"Video3DFormat":null,"MediaStreams":[{"Codec":"hevc","CodecTag":null,"Language":null,"ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"DvVersionMajor":null,"DvVersionMinor":null,"DvProfile":null,"DvLevel":null,"RpuPresentFlag":null,"ElPresentFlag":null,"BlPresentFlag":null,"DvBlSignalCompatibilityId":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":null,"VideoRange":"SDR","VideoRangeType":"SDR","VideoDoViTitle":null,"LocalizedUndefined":null,"LocalizedDefault":null,"LocalizedForced":null,"LocalizedExternal":null,"DisplayTitle":"1080p HEVC SDR","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":null,"BitRate":1799185,"BitDepth":8,"RefFrames":1,"PacketLength":null,"Channels":null,"SampleRate":null,"IsDefault":true,"IsForced":false,"Height":1080,"Width":1920,"AverageFrameRate":60,"RealFrameRate":60,"Profile":"Main","Type":1,"AspectRatio":"16:9","Index":0,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":"yuv420p","Level":123,"IsAnamorphic":null},{"Codec":"eac3","CodecTag":null,"Language":null,"ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"DvVersionMajor":null,"DvVersionMinor":null,"DvProfile":null,"DvLevel":null,"RpuPresentFlag":null,"ElPresentFlag":null,"BlPresentFlag":null,"DvBlSignalCompatibilityId":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":"GPAC ISO Audio Handler","VideoRange":null,"VideoRangeType":null,"VideoDoViTitle":null,"LocalizedUndefined":null,"LocalizedDefault":null,"LocalizedForced":null,"LocalizedExternal":null,"DisplayTitle":"GPAC ISO Audio Handler - Dolby Digital\u002B - Stereo - Default","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"stereo","BitRate":192000,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":2,"SampleRate":48000,"IsDefault":true,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":0,"AspectRatio":null,"Index":1,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null},{"Codec":"eac3","CodecTag":null,"Language":null,"ColorRange":null,"ColorSpace":null,"ColorTransfer":null,"ColorPrimaries":null,"DvVersionMajor":null,"DvVersionMinor":null,"DvProfile":null,"DvLevel":null,"RpuPresentFlag":null,"ElPresentFlag":null,"BlPresentFlag":null,"DvBlSignalCompatibilityId":null,"Comment":null,"TimeBase":"1/1000","CodecTimeBase":null,"Title":"GPAC ISO Audio Handler","VideoRange":null,"VideoRangeType":null,"VideoDoViTitle":null,"LocalizedUndefined":null,"LocalizedDefault":null,"LocalizedForced":null,"LocalizedExternal":null,"DisplayTitle":"GPAC ISO Audio Handler - Dolby Digital\u002B - 5.1 - Default","NalLengthSize":null,"IsInterlaced":false,"IsAVC":null,"ChannelLayout":"5.1","BitRate":448000,"BitDepth":null,"RefFrames":null,"PacketLength":null,"Channels":6,"SampleRate":48000,"IsDefault":true,"IsForced":false,"Height":null,"Width":null,"AverageFrameRate":null,"RealFrameRate":null,"Profile":null,"Type":0,"AspectRatio":null,"Index":2,"Score":null,"IsExternal":false,"DeliveryMethod":null,"DeliveryUrl":null,"IsExternalUrl":null,"IsTextSubtitleStream":false,"SupportsExternalStream":false,"Path":null,"PixelFormat":null,"Level":0,"IsAnamorphic":null}],"MediaAttachments":[],"Formats":[],"Bitrate":2439185,"Timestamp":null,"RequiredHttpHeaders":{},"TranscodingUrl":null,"TranscodingSubProtocol":null,"TranscodingContainer":null,"AnalyzeDurationMs":null,"DefaultAudioStreamIndex":null,"DefaultSubtitleStreamIndex":null}

/usr/lib/jellyfin-ffmpeg/ffmpeg -analyzeduration 200M -ss 00:00:21.773 -fflags +genpts  -i file:"/mnt/share/data/media/series/TEST TDARR/Season 1000/sample__1080__libx265__eac3__30s__video (1).mkv" -map 0:0 -map 0:1 -map -0:s -codec:v:0 copy -map_metadata -1 -map_chapters -1 -threads 0 -codec:a:0 libfdk_aac -ac 2 -ab 384000 -f mp4 -movflags frag_keyframe+empty_moov -y "/var/lib/jellyfin/transcodes/ca5df9fc4a63c1d06c1d29972f2b5e34.mp4"


ffmpeg version 5.1.2-Jellyfin Copyright (c) 2000-2022 the FFmpeg developers
  built with gcc 11 (Ubuntu 11.3.0-1ubuntu1~22.04)
  configuration: --prefix=/usr/lib/jellyfin-ffmpeg --target-os=linux --extra-libs=-lfftw3f --extra-version=Jellyfin --disable-doc --disable-ffplay --disable-ptx-compression --disable-shared --disable-libxcb --disable-sdl2 --disable-xlib --enable-lto --enable-gpl --enable-version3 --enable-static --enable-gmp --enable-gnutls --enable-chromaprint --enable-libdrm --enable-libass --enable-libfreetype --enable-libfribidi --enable-libfontconfig --enable-libbluray --enable-libmp3lame --enable-libopus --enable-libtheora --enable-libvorbis --enable-libdav1d --enable-libwebp --enable-libvpx --enable-libx264 --enable-libx265 --enable-libzvbi --enable-libzimg --enable-libfdk-aac --arch=amd64 --enable-libsvtav1 --enable-libshaderc --enable-libplacebo --enable-vulkan --enable-opencl --enable-vaapi --enable-amf --enable-libmfx --enable-ffnvcodec --enable-cuda --enable-cuda-llvm --enable-cuvid --enable-nvdec --enable-nvenc
  libavutil      57. 28.100 / 57. 28.100
  libavcodec     59. 37.100 / 59. 37.100
  libavformat    59. 27.100 / 59. 27.100
  libavdevice    59.  7.100 / 59.  7.100
  libavfilter     8. 44.100 /  8. 44.100
  libswscale      6.  7.100 /  6.  7.100
  libswresample   4.  7.100 /  4.  7.100
  libpostproc    56.  6.100 / 56.  6.100
Input #0, matroska,webm, from 'file:/mnt/share/data/media/series/TEST TDARR/Season 1000/sample__1080__libx265__eac3__30s__video (1).mkv':
  Metadata:
    title           : Big Buck Bunny, Sunflower version
    GENRE           : Animation
    MAJOR_BRAND     : isom
    MINOR_VERSION   : 1
    COMPATIBLE_BRANDS: isomavc1
    COMPOSER        : Sacha Goedegebure
    ARTIST          : Blender Foundation 2008, Janus Bager Kristensen 2013
    COMMENT         : Creative Commons Attribution 3.0 - http://bbb3d.renderfarming.net
    ENCODER         : Lavf57.83.100
  Duration: 00:00:30.02, start: 0.000000, bitrate: 1799 kb/s
  Stream #0:0: Video: hevc (Main), yuv420p(tv, progressive), 1920x1080 [SAR 1:1 DAR 16:9], 60 fps, 60 tbr, 1k tbn (default)
    Metadata:
      HANDLER_NAME    : GPAC ISO Video Handler
      ENCODER         : Lavc57.107.100 libx265
      DURATION        : 00:00:30.005000000
  Stream #0:1: Audio: eac3, 48000 Hz, stereo, fltp, 192 kb/s (default)
    Metadata:
      HANDLER_NAME    : GPAC ISO Audio Handler
      ENCODER         : Lavc57.107.100 eac3
      DURATION        : 00:00:30.016000000
  Stream #0:2: Audio: eac3, 48000 Hz, 5.1(side), fltp, 448 kb/s (default)
    Metadata:
      HANDLER_NAME    : GPAC ISO Audio Handler
      ENCODER         : Lavc57.107.100 eac3
      DURATION        : 00:00:30.016000000
Stream mapping:
  Stream #0:0 -> #0:0 (copy)
  Stream #0:1 -> #0:1 (eac3 (native) -> aac (libfdk_aac))
Press [q] to stop, [?] for help
Output #0, mp4, to '/var/lib/jellyfin/transcodes/ca5df9fc4a63c1d06c1d29972f2b5e34.mp4':
  Metadata:
    encoder         : Lavf59.27.100
  Stream #0:0: Video: hevc (Main) (hev1 / 0x31766568), yuv420p(tv, progressive), 1920x1080 [SAR 1:1 DAR 16:9], q=2-31, 60 fps, 60 tbr, 16k tbn (default)
  Stream #0:1: Audio: aac (mp4a / 0x6134706D), 48000 Hz, stereo, s16, 384 kb/s (default)
    Metadata:
      encoder         : Lavc59.37.100 libfdk_aac
frame=    4 fps=0.0 q=-1.0 size=       0kB time=00:00:00.00 bitrate=N/A speed=   0x    
frame=  592 fps=0.0 q=-1.0 Lsize=    1652kB time=00:00:08.25 bitrate=1639.2kbits/s speed=43.2x    
video:1250kB audio:389kB subtitle:0kB other streams:0kB global headers:2kB muxing overhead: 0.813894%
```

### Please attach any browser or client logs here

no errors

### Please attach any screenshots here

![brave_6gwNnksopM](https://user-images.githubusercontent.com/40745505/219458787-b9f0496f-c9e0-44ff-9f30-0c7eb19beea2.png)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by pCoder3000** [comment]
thanks for writing the issues i have the same problem and can only add that the subtitle is also out of sync. In my tests the sound and the subtitle were about 5 seconds apart.

**Comment by xZeroKooLx** [comment]
I've only been using jellyfin the past week and was hesitant to report this as a bug as I thought it must have been an issue with my setup but I am seeing the exact same results under the same circumstances.

**Comment by Mixzzz** [comment]
I have the exact same issue. I have a collection of family guy in h265 and if I skip, the audio becomes desynced. I also transcoded a movie today with tdarr and while direct playing H265 and AAC audio, its desynced. When I switch to lower bitrate forcing it to transcode, everything is fine. Everything is fine also for H264 with either direct streaming or transcoding.

**Comment by arjunsrinivasan1997** [comment]
I'm also having this issue. Weirdly enough if I use the video player app it works fine, but if i try and watch via just navigating to my server URL in a browser stuff is very out of sync. Forcing a transcoded versions, instead of a direct streamed one also fixes it.

**Comment by nicolai455** [comment]
Im pretty sure its some weird issue with chromium hevc (x265) support, the same sample file in avc (x264) has no issues

**Comment by DieserMerlin** [comment]
I have the same problem, also with h265 only.

**Comment by BushTrocity** [comment]
I thought I was going mad. I have the same issue after transcoding all my media to hevc.

**Comment by Aidan647** [comment]
I have tested it with any non direct audio tested on EAC3/AC3/DTS

**Comment by Bond-009** [comment]
Duplicate of #9241

**Comment by StaZhu** [comment]
> Im pretty sure its some weird issue with hevc (h265) support, the same sample file in avc (h264) has no issues

Please fire a bug to crbug.com if you think this is a browser implement issue for H265.

Also to note can you reproduce this issue on other server like plex?

**Comment by SirJMD** [comment]
Same issue here. Works fine in Emby 4.7.11.0.

**Comment by pascal-de-ladurantaye** [comment]
Adding my quick 2 cents: Seems to be a Chromium only issue as I'm not able to reproduce in Firefox but I can easily reproduce in Chrome

**Comment by jackhric** [comment]
+1
Having this issue as well, and like @pascal-de-ladurantaye said, using Firefox or Microsoft Edge (native H265 not implemented yet?) fixes the issue. Posted about it on the subreddit thinking it might have just been me, but turns out quite a bit of us are having the issue too. Here's my tests and play logs:

Case 1: 28 Days Later

[https://streamable.com/m0otq0](https://streamable.com/m0otq0)
[https://pastebin.com/kM3UkWAg](https://pastebin.com/kM3UkWAg)

Case 2: Blade (1998)

[https://streamable.com/y69qvh](https://streamable.com/y69qvh)
[https://i.imgur.com/lumz4yc.png](https://i.imgur.com/lumz4yc.png)
[https://pastebin.com/kpduTZmF](https://pastebin.com/kpduTZmF)

**Comment by Matix-Media** [comment]
Same issue. Limiting audio to 7.1 on desktop chrome fixes the issue for me, but only desktop, not mobile android app (also chrome based?)

**Comment by gionag** [comment]
+1 ! when only audio is transcoded on chrome web and android client the audio is out of sync

**Comment by mwdiers** [comment]
It's not just H.265. I have this problem on AV1 10-bit on Chrome.

**Comment by nullfact0r** [comment]
Same issue here, just that audio isn't just out of sync - if I don't watch something from beginning to end it just stops working altogether

**Comment by Bond-009** [comment]
Please top commenting in this closed issue, instead go here: https://github.com/jellyfin/jellyfin/issues/9241
